src/Components/App/App.tsx

```diff
@@ -21,31 +21,43 @@ function App() {
     getCreators()
       .then(data => {
         setAllCreators(data.data.attributes.creators);
-        setDisplayedCreators(data.data.attributes.creators);
       })
       .catch(error => setError('Failed to get creators. Please try again later.'));
-
     getUserInfo()
       .then(data => {
         setUser(data.data.attributes);
         setSavedCreators(data.data.attributes.follows);
       })
-      .catch(error => setError('Failed to display user info. Please try again.'));
+      .catch(error => setError('Failed to display user info. Please try again later.'));
   }, []);
 
   const handlePageSwitch = (tab: string) => {
-    if (tab === 'all' || tab === 'saved' || tab === 'home') {
-      setActiveTab(tab as 'saved' | 'all' | 'home');
       if (tab === 'all') {
         setDisplayedCreators(allCreators);
       } else if (tab === 'saved') {
         setDisplayedCreators(savedCreators);
       } else {
         setDisplayedCreators(null);
-      }
-    }
-  }
-  
+      };
+  };
+
+  const favoriteCreator = (creatorInfo: CreatorInfo) => {
```

Comment:
Great usage of parsing here for the individual creators unique Id for saving into our saved creators tab. I was unsure how to go about this myself in prior projects and this gives good insight on how to make that possible!